### PR TITLE
Add custom release instructions

### PR DESCRIPTION
**Pull request by sjezewski**
Distilling this knowledge while its fresh.

### CODE REVIEW

File: doc/release_instructions.md [hunk 1]
```diff
@@ -6,6 +6,8 @@ This is because of a dependency on CGO via [this bug](https://github.com/opencon
 
 (We don't want to enable CGO in part because it doesn't play nice w OSX for us)
 
+If you're doing a custom release (off a branch that isn't master, skip to the section at the bottom)
```
Comment (gabrielgrant) [inline doc/release_instructions.md]: I think the close-bracket should be before the comma here, right? Also, might as well add a link:

```
If you're doing a custom release (off a branch that isn't master), skip to [the section at the bottom](#custom-release)
```

File: doc/release_instructions.md [hunk 2]
```diff
@@ -52,4 +54,19 @@ make point-release
...
+## Custom Release
+
+Occasionally we have a need for a custom release off a non master branch. This is usually because some features we need to supply to users that are incompatible w features on master, but the features on master we need to keep longer term.
+
+Follow the instructions above, just run the make script off of your branch. 
+
+Then after a successful release:
+
+- The tag created by goxc will point to master, and this is wrong. Opened an issue for this: https://github.com/laher/goxc/issues/112
+  - To mitigate this, you'll have to go to github.com/pachyderm/pachyderm/releases, and manually delete the release that goxc created
```
Comment (gabrielgrant) [inline doc/release_instructions.md]: add `https://` to `github.com/pachyderm/pachyderm/releases` so autolinking works

**Comment by sjezewski** [comment]
Cool. Made those changes

**Comment by sjezewski** [comment]
Yup @gg waiting on a +1 from you before I click the merge button.

**Comment by gabrielgrant** [comment]
LGTM

**Comment by gabrielgrant** [comment]
This should be updated to include instructions for generating binary build artifacts once #1134 is fixed

**Comment by sjezewski** [comment]
Oh man, I came here to make that same comment. You're a step ahead! Nice!

**Comment by sjezewski** [comment]
Ok. Updated. Please take another look

**Comment by gabrielgrant** [comment]
Ok, now it LGTM as well as Trump. He's not much of a detail guy ;)

**Pull request merged**